migrations/000005_partition_functions.up.sql

```diff
@@ -0,0 +1,58 @@
+CREATE OR REPLACE FUNCTION create_range_patition(partition_table_name TEXT, parent_table Text, partition_start_date Text, partition_end_date Text)
+RETURNS void AS
+$BODY$
+DECLARE
+BEGIN
+   EXECUTE 'CREATE TABLE ' || partition_table_name
+      || ' PARTITION OF '|| parent_table ||' FOR VALUES FROM '
+      || '(''' || partition_start_date || ''')'
+      || ' TO ' 
+      || '(''' || partition_end_date || ''')';
+END;
+$BODY$
+LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION create_monthly_patitions(max_interval_end TIMESTAMP WITH TIME ZONE, parent_table Text)
+RETURNS void AS
+$BODY$
+DECLARE
+   record_day INT;
+   record_date TEXT;
+   partition_start_date TEXT;
+   partition_end_date TEXT;
+   partition_table_name TEXT; 
+BEGIN
+    record_day := TO_NUMBER(TO_CHAR(max_interval_end,'DD'),'99');
+    record_date := TO_CHAR(max_interval_end,'YYYY-MM-');
+    IF record_day > 15 THEN
+        partition_start_date = CONCAT(record_date, '16');
+        select (date_trunc('month', max_interval_end) + interval '1 month - 1 day')::date INTO partition_end_date;
```

Comment:
Note: Partition end_date should be start of the next month and not end of current month.

Reply:
Fixed.